modules/kafka/src/main/java/org/testcontainers/containers/KafkaContainer.java

```diff
@@ -42,50 +57,145 @@ public KafkaContainer(final DockerImageName dockerImageName) {
+            throw new IllegalStateException("Cannot configure Kraft mode when Zookeeper configured");
+        }
+        String actualVersion = DockerImageName.parse(getDockerImageName()).getVersionPart();
+        if (new ComparableVersion(actualVersion).isLessThan(MIN_KRAFT_TAG)) {
+            throw new IllegalArgumentException(
+                String.format(
+                    "Provided Confluent Platform's version %s is not supported in Kraft mode (must be %s or above)",
+                    actualVersion,
+                    MIN_KRAFT_TAG
+                )
+            );
+        }
+        kraftEnabled = true;
+        return self();
+    }
+
+    public KafkaContainer withClusterId(String clusterId) {
+        this.clusterId = clusterId;
+        return self();
+    }
+
     public String getBootstrapServers() {
         return String.format("PLAINTEXT://%s:%s", getHost(), getMappedPort(KAFKA_PORT));
     }
 
+    public static String generateClusterId() {
+        UUID metadataTopicUuid = new UUID(0, 1L);
+        UUID ZeroUuid = new UUID(0, 0L);
+
+        UUID candidate = UUID.randomUUID();
+        while(candidate.equals(metadataTopicUuid) || candidate.equals(ZeroUuid)) {
+            candidate = UUID.randomUUID();
+        }
+
+        //From Java UUID to Kafka Uuid representation
+        ByteBuffer uuidBytes = ByteBuffer.wrap(new byte[16]);
+        uuidBytes.putLong(candidate.getMostSignificantBits());
+        uuidBytes.putLong(candidate.getLeastSignificantBits());
+        final byte[] uuidBytesArray = uuidBytes.array();
+
+        return Base64.getUrlEncoder().withoutPadding().encodeToString(uuidBytesArray);
+
+    }
+
     @Override
     protected void configure() {
-        withEnv(
-            "KAFKA_ADVERTISED_LISTENERS",
-            String.format("BROKER://%s:9092", getNetwork() != null ? getNetworkAliases().get(0) : "localhost")
-        );
+        if (kraftEnabled) {
+            waitingFor(Wait.forLogMessage(".*Transitioning from RECOVERY to RUNNING.*", 1));
+        } else {
+            waitingFor(Wait.forLogMessage(".*\\[KafkaServer id=\\d+\\] started.*", 1));
+        }
 
-        String command = "#!/bin/bash\n";
-        if (externalZookeeperConnect != null) {
+        if (kraftEnabled) {
+            withEnv(
+                "KAFKA_LISTENER_SECURITY_PROTOCOL_MAP",
+                String.format("%s,CONTROLLER:PLAINTEXT", getEnvMap().get("KAFKA_LISTENER_SECURITY_PROTOCOL_MAP"))
+            );
+            withEnv(
+                "KAFKA_LISTENERS",
+                String.format("%s,CONTROLLER://0.0.0.0:9094", getEnvMap().get("KAFKA_LISTENERS"))
+            );
+
+            withEnv("KAFKA_PROCESS_ROLES", "broker,controller");
+            if(!getEnvMap().containsKey("KAFKA_CONTROLLER_QUORUM_VOTERS")) {
+                withEnv(
+                    "KAFKA_CONTROLLER_QUORUM_VOTERS",
+                    String.format("%d@%s:9094",
+                        getEnvMap().get("KAFKA_NODE_ID"),
```

Comment:
Besides the format, `%d` is updated to `%s`

```suggestion
                    String.format(
                        "%s@%s:9094",
                        getEnvMap().get("KAFKA_NODE_ID"),
```